plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -4131,13 +4131,19 @@ public void testAllAvailableTypes()
     @Test(dataProvider = "repartitioningDataProvider")
     public void testRepartitionDataOnCtas(Session session, String partitioning, int expectedFiles)
     {
-        testRepartitionData(session, "tpch.tiny.orders", true, partitioning, expectedFiles);
+        // Lowercase `comment` column which is used for partitioning. Otherwise, cleanup routine which deletes unnecessary files after INSERT/CTAS may
+        // delete unexpected files when executed on case-insensitive filesystem; this is a test environment (non-production) issue.
+        String sourceRelation = "(SELECT orderkey, custkey, orderstatus, totalprice, orderdate, orderpriority, clerk, shippriority, lower(comment) AS comment FROM tpch.tiny.orders)";
```

Comment:
Another approach would be to change [cleanExtraOutputFiles](https://github.com/trinodb/trino/blob/a0e06e426c75c0618470453b5c646c60a7822487/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L826), so it strips location from path stored in `filesToKeep`. Theoretically, it may prevent us from deleting something we should delete, but it is a very slim chance of that. No chance really as filenames contain UUID.

@findepi do you think I should drop the change in the test and modify prod code instead?

Reply:
> Another approach would be to change [cleanExtraOutputFiles](https://github.com/trinodb/trino/blob/a0e06e426c75c0618470453b5c646c60a7822487/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L826), so it strips location from path stored in `filesToKeep`.


That's what @alexjo2144 did for orphan removal.
It's because paths can be compared reliably only if they are "canonical". But the concept of canonical paths is not very well defined, is it?

Sounds reasonable to me.